src/FilesystemRegistry.php

```diff
@@ -173,6 +202,36 @@ protected static function adapter($adapter, array $vars)
         throw new \InvalidArgumentException('Unknown adapter');
     }
 
+    /**
+     * Parse dsn, map configuration and instantiate adapter.
+     *
+     * @param string $dsn Data Source Name to parse and map.
+     * 
+     * @throws \InvalidArgumentException Thrown when the given adapter class doesn't exists.
+     *
+     * @return AdapterInterface
+     */
+    protected static function adapterFromDsn(string $dsn){
+        $vars = static::parseDsn($dsn);
+
+        if(class_exists($vars['className'])){
+            throw new \InvalidArgumentException('Unknown adapter');
+        }
+
+        if($vars['className'] == 'League\Flysystem\AwsS3v3\AwsS3v3Adapter'){
```

Comment:
Since this is hard coded it makes more sense IMHO to put factory classes in `$_dsnClassMap` and put the custom logic in there.

Reply:
ok, added that on top ;) please review

Reply:
:+1: